kv/etcd/mock_test.go

```diff
@@ -0,0 +1,394 @@
+		res, err := kv.Get(context.Background(), "/foo")
+
+		require.NoError(t, err)
+		require.Len(t, res.Kvs, 1)
+		assert.Equal(t, []byte("/foo"), res.Kvs[0].Key)
+	})
+
+	t.Run("not exact match", func(t *testing.T) {
+		pair := mvccpb.KeyValue{
+			Key:   []byte("/foo"),
+			Value: []byte("1"),
+		}
+
+		kv := newMockKV()
+		kv.values = map[string]mvccpb.KeyValue{string(pair.Key): pair}
+		res, err := kv.Get(context.Background(), "/bar")
+
+		require.NoError(t, err)
+		assert.Empty(t, res.Kvs)
+	})
+
+	t.Run("prefix match", func(t *testing.T) {
+		pair1 := mvccpb.KeyValue{
+			Key:   []byte("/foo"),
+			Value: []byte("1"),
+		}
+		pair2 := mvccpb.KeyValue{
+			Key:   []byte("/baz"),
+			Value: []byte("2"),
+		}
+		pair3 := mvccpb.KeyValue{
+			Key:   []byte("/first"),
+			Value: []byte("3"),
+		}
+
+		kv := newMockKV()
+		kv.values = map[string]mvccpb.KeyValue{
+			string(pair1.Key): pair1,
+			string(pair2.Key): pair2,
+			string(pair3.Key): pair3,
+		}
+		res, err := kv.Get(context.Background(), "/f", clientv3.WithPrefix())
+
+		require.NoError(t, err)
+		assert.ElementsMatch(t, []*mvccpb.KeyValue{&pair1, &pair3}, res.Kvs)
+	})
+
+	t.Run("empty prefix", func(t *testing.T) {
+		pair1 := mvccpb.KeyValue{
+			Key:   []byte("/foo"),
+			Value: []byte("1"),
+		}
+
+		kv := newMockKV()
+		kv.values = map[string]mvccpb.KeyValue{string(pair1.Key): pair1}
+		res, err := kv.Get(context.Background(), "", clientv3.WithPrefix())
+
+		require.NoError(t, err)
+		assert.ElementsMatch(t, []*mvccpb.KeyValue{&pair1}, res.Kvs)
+	})
+}
+
+func TestMockKV_Put(t *testing.T) {
+	t.Run("new key", func(t *testing.T) {
+		kv := newMockKV()
+		_, err := kv.Put(context.Background(), "/foo", "1")
+
+		require.NoError(t, err)
+		assert.Equal(t, int64(1), kv.values["/foo"].Version)
+		assert.Equal(t, []byte("1"), kv.values["/foo"].Value)
+	})
+
+	t.Run("existing key", func(t *testing.T) {
+		kv := newMockKV()
+		kv.values["/foo"] = mvccpb.KeyValue{
+			Key:            []byte("/foo"),
+			CreateRevision: 1,
+			ModRevision:    2,
+			Version:        2,
+			Value:          []byte("1"),
+		}
+
+		_, err := kv.Put(context.Background(), "/foo", "2")
+
+		require.NoError(t, err)
+		assert.Equal(t, int64(3), kv.values["/foo"].Version)
+		assert.Equal(t, []byte("2"), kv.values["/foo"].Value)
+	})
+}
+
+func TestMockKV_Delete(t *testing.T) {
+	t.Run("exact match", func(t *testing.T) {
+		kv := newMockKV()
+		kv.values["/foo"] = mvccpb.KeyValue{
+			Key:   []byte("/foo"),
+			Value: []byte("1"),
+		}
+
+		res, err := kv.Delete(context.Background(), "/foo")
+
+		require.NoError(t, err)
+		assert.Equal(t, int64(1), res.Deleted)
+		assert.Empty(t, kv.values)
+	})
+
+	t.Run("prefix match", func(t *testing.T) {
+		pair1 := mvccpb.KeyValue{
+			Key:   []byte("/foo"),
+			Value: []byte("1"),
+		}
+		pair2 := mvccpb.KeyValue{
+			Key:   []byte("/baz"),
+			Value: []byte("2"),
+		}
+		pair3 := mvccpb.KeyValue{
+			Key:   []byte("/first"),
+			Value: []byte("3"),
+		}
```

Comment:
same as above with the rename suggestions to `fooPair`, `bazPair`, and `firstPair`